framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceJavaParserStorage.java

```diff
@@ -1146,6 +1222,21 @@ public boolean addDeclarationAnnotation(AnnotationMirror annotation) {
       return declarationAnnotations.add(annotation);
     }
 
+    /**
+     * Adds a declaration annotation to this parameter and returns whether it was a new annotation.
+     *
+     * @param annotation declaration annotation to add
+     * @param index index of the parameter
+     * @return true if {@code annotation} wasn't previously stored for this parameter
+     */
+    public boolean addDeclarationAnnotationForParams(AnnotationMirror annotation, int index) {
```

Comment:
I would be inclined to drop the trailing `s` because this adds an annotation to exactly one parameter.

Or, should the names in this file be more consistent with those in `WholeProgramInferenceScenesStorage.java`?  That could improve naming and understandability.